include/deal.II/lac/relaxation_block.h

```diff
@@ -147,10 +147,23 @@ class RelaxationBlock :
      * </ol>
      */
     std::vector<std::vector<unsigned int> > order;
+
     /**
      * Return the memory allocated in this object.
      */
     std::size_t memory_consumption() const;
+
+    /**
+     * Temporary ghost vector that is used in the Relaxation method when
+     * performing parallel MPI computations. The user is required to have this
+     * point to an initialized vector that contains at least the locally
+     * active level DoFs.
```

Comment:
> Ah, it finally becomes clear to me how this is supposed to work -- this is a public variable, rather than something the class uses internally. Have you thought about whether there are alternatives?

Yes. I wrote above: "The hack to specify the ghost vector is somewhat ugly, but 1. RelaxationBlock doesn't know the vector type (this is a template argument to the step function only), 2. RelaxationBlock knows nothing about the DoFHandler so creating the ghost vector inside is not possible (or not efficient). Sadly, I don't know of a better way to do this."

I have thought about it some more and I realized that the current approach becomes quite ugly if you consider we want to support PETSc (or other) vector formats too. Maybe we need to make the vector type a template parameter of the class. Thoughts?

@guidokanschat What is your thought on this?